File: Volume_POO.py
class Esfera:
    def __init__(self, pi, r):
        self.pi = pi
        self.r = r
    
    def volume(self):
        volume = 4 * self.pi * self.r * self.r  * self.r  * self.r / 3
        print(volume)

class Piramide:
    def __init__(self, area_base, h):
        self.area_base = area_base
        self.h = h
    
    def volume(self):
        volume = self.area_base * self.h / 3
        print(volume)


class Cilindro:
    def __init__(self, pi, r, h):
        self.pi = pi
        self.r = r
        self.h = h
    
    def volume(self):
        volume = self.pi * self.r * self.r  * self.h
        print(volume)
        
e = Esfera(3.14, 10)
p = Piramide(10, 15)
c = Cilindro(3.14, 15, 10)

e.volume()
p.volume()
c.volume()

File: Volume_POO.c
#include <stdio.h>

 typedef struct {
        float pi;
        float r;
    } Esfera;

    typedef struct {
        float area_base;
        float h;
    } Piramide;
    
    typedef struct {
        float pi;
        float r;
        float h;
    } Cilindro;


    void volume_esfera(float r, float pi) {
        float volume = 4 * pi * r * r * r/ 3;
        printf("%f \n", volume);
    }

    void volume_piramide(float area_base, float h) {
        float volume = area_base * h / 3;
        printf("%f \n", volume);
    }
    void volume_cilindro(float r, float pi, float h) {
        float volume = pi * r * r * h;
        printf("%f \n", volume);
    }

int main() {

   Esfera e;
   Piramide p;
   Cilindro c;

   e.pi = 3.14;
   c.pi = 3.14;

   e.r = 10;
   p.area_base = 5;
   p.h = 7;
   c.r = 10;
   c.h = 12;

   
   volume_esfera(e.r, e.pi);
   volume_piramide(p.area_base, p.h);
   volume_cilindro(c.r, c.pi, c.h);




    return 0;
}
